[MOSIP-34747] - set spring.profiles.active composite (#1553)

swatigoel opened the pull request:
spring active profile is set to composite.
To make it work, we need to add composite properties through env variables.
Adding type and uri are mandatory properties.
If we don't specify default_label which is known as branch, it will refer to main branch of repo. In our repositories we have main branch as master. Because of this providing default_label becomes mandatory for us.

Env variables to set for composite profile are as
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_URI=https://github.com/mosip/inji-config
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_TYPE=git
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_0_DEFAULT_LABEL=develop

export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_URI=https://github.com/mosip/mosip-config
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_TYPE=git
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_DEFAULT_LABEL=develop

for native
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_URI=file-path-for-local-properties
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_TYPE=native
export SPRING_CLOUD_CONFIG_SERVER_COMPOSITE_1_DEFAULT_LABEL=develop

swatigoel committed:
[MOSIP-34747] - set spring.profiles.active composite (#1553)

-- Update readme with example to support different type of repositories - git, svn, native

Signed-off-by: Swati Goel <meet2swati@gmail.com>

vishwa-vyom reviewed (approved):
since this from a feature branch, which was previously reviewed, approving this with a quick check.

swatigoel committed:
Sync with develop (#1570)

* fix: kernel/kernel-idobjectvalidator/pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236
- https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-1015415
- https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-5710356

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* fix: kernel/kernel-bom/pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-ORGJSON-5488379
- https://snyk.io/vuln/SNYK-JAVA-ORGJSON-5962464

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* fix: kernel/kernel-pdfgenerator-itext/pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-COMITEXTPDF-2320389
- https://snyk.io/vuln/SNYK-JAVA-ORGBOUNCYCASTLE-1052448
- https://snyk.io/vuln/SNYK-JAVA-ORGBOUNCYCASTLE-2841508
- https://snyk.io/vuln/SNYK-JAVA-COMITEXTPDF-541282
- https://snyk.io/vuln/SNYK-JAVA-COMITEXTPDF-541284
- https://snyk.io/vuln/SNYK-JAVA-COMITEXTPDF-2391484
- https://snyk.io/vuln/SNYK-JAVA-COMITEXTPDF-2391485
- https://snyk.io/vuln/SNYK-JAVA-COMITEXTPDF-2391486

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* Revert "fix: kernel/kernel-pdfgenerator-itext/pom.xml to reduce vulnerabilities"

This reverts commit 80a480eeefe55ae0b4d64c2da017a26a0d5ee64e.

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* fix: kernel/kernel-authcodeflowproxy-api/pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-ORGAPACHETOMCATEMBED-6435950
- https://snyk.io/vuln/SNYK-JAVA-ORGHIBERNATEVALIDATOR-541187
- https://snyk.io/vuln/SNYK-JAVA-ORGHIBERNATEVALIDATOR-6247635
- https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-467268
- https://snyk.io/vuln/SNYK-JAVA-ORGHIBERNATEVALIDATOR-568163
- https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-5710356

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* fix: kernel/kernel-bom/pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-ORGAPACHETOMCATEMBED-7430175
- https://snyk.io/vuln/SNYK-JAVA-COMMONSIO-1277109
- https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORKSECURITY-6457293
- https://snyk.io/vuln/SNYK-JAVA-IONETTY-6483812
- https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-6444790
- https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-6597980
- https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHTTPCOMPONENTS-1048058

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* Update pom.xml

Signed-off-by: rajapandi1234 <138785181+rajapandi1234@users.noreply.github.com>
Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* Update pom.xml

Signed-off-by: rajapandi1234 <138785181+rajapandi1234@users.noreply.github.com>
Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* Update pom.xml

Signed-off-by: rajapandi1234 <138785181+rajapandi1234@users.noreply.github.com>
Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>

* Update pom.xml

Signed-off-by: rajapandi1234 <138785181+rajapandi1234@users.noreply.github.com>

* [DSD-6020] Updated nexus-staging-maven-plugin version in pom.xml

Signed-off-by: techno-467 <prafulrakhade02@gmail.com>

* [MOSIP-35226] Enabled config-server to pull configurations from local git repo. (#1566)

* [MOSIP-35226] Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* [MOSIP-35226] Updated Dockerfile

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated Dockerfile

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* Updated Dockerfile

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* [MOSIP-35226] Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* [MOSIP-35226] Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* [MOSIP-35226] Updated bootstrap.properties

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

---------

Signed-off-by: Mohan E <mohanraj1715@gmail.com>

* [MOSIP-35160] Updated URL from https://github.com/mosip/mosip-infra/blob/master/deployment/v3/utils/copy_cm_func.sh to https://raw.githubusercontent.com/mosip/mosip-infra/master/deployment/v3/utils/copy_cm_func.sh

Signed-off-by: techno-467 <prafulrakhade02@gmail.com>

---------

Signed-off-by: rajapandi.m <rajapandi.m@technoforte.co.in>
Signed-off-by: rajapandi1234 <138785181+rajapandi1234@users.noreply.github.com>
Signed-off-by: techno-467 <prafulrakhade02@gmail.com>
Signed-off-by: Mohan E <mohanraj1715@gmail.com>
Co-authored-by: snyk-bot <snyk-bot@snyk.io>
Co-authored-by: rajapandi1234 <138785181+rajapandi1234@users.noreply.github.com>
Co-authored-by: Chandra Keshav Mishra <chandrakeshavmishra@gmail.com>
Co-authored-by: techno-467 <prafulrakhade02@gmail.com>
Co-authored-by: Mohan E <mohanraj1715@gmail.com>

swatigoel commented:
raising another PR by resolving conflicts
